### ISSUE TITLE
"Scanner could not load film"

### ISSUE THREAD

**Issue by sloaneross** [kosma/coolscan-mods#3]
Not sure where the fault lies here, either with the firmware or some physical issue related to the cutting of the hole in the back. 

I have a LS-50 ED, I have a SA-21. I followed the instructions and I can get the scanner to correctly identify that it *can* scan 40 frames. But, there's a problem. 

In NikonScan any attempt to scan it pulls the film in, then ejects it with the error _The scanner could not load the film, please consult your users manual _

In VueScan trying to preview it scans about 1mm of the first frame, and uses that as the preview for everything. I'm a bit at a loss as to what is going on.

Attached are the memory dump and the error I get from nikon scan
![Screenshot 2023-06-21 230339](https://github.com/kosma/coolscan-mods/assets/74197030/9426dc2b-8402-4dde-ae47-8d9eb4c8fc85)
![Screenshot 2023-06-21 230435](https://github.com/kosma/coolscan-mods/assets/74197030/08c40ad9-9186-4a29-9628-22d88c8e81f3)

**Comment by sloaneross** [comment]
Update, I have gotten it working with vuescan but this software is bad. I'm not sure how but I would like to try to work on getting it working with nikonscan

EDIT: on vuescan it will only scan 6 frames

**Comment by Toesoe** [comment]
Same issue here on my LS-50 ED. Vuescan seems to work fine, but Nikon Scan refuses to load the strip, it spits it out after trying to obtain either a preview or a 6-frame scan. A 2-frame scan seems to work fine, though.
I haven't tested 6+ frames yet as I didn't cut a hole.

Dump:

```
***************************************
* Nikon Memory Dumper by Kosma Moczek *
***************************************
Connecting to scanner \\.\usbscan0...
...connected.
Checking model: 0x04b0, product: 0x4001, version: 2.02
...model: DF17811
Checking scanner...
cmd: 12 00 00 00 24 00
...scanner says it's a Nikon   LS-50 ED        2.02
Reading memory at 0xffffc0 to 0xffffdf...
cmd: 3c 02 00 ff ff c0 00 00 20 00
I/O Registers:
0000: ff ff 00 00 ff ff 01 00
0008: ff ff 00 ff ff ff ff eb
0010: ff ff fc 3f ff ff 00 d7
0018: 00 ff 00 00 00 00 1f ff

Finished.
Press any key to continue...
```

I have a feeling this is related to Nikon Scan. Maybe it expects a sensor value returned from the scanner from the film gate at the back when you have 6+ frames on a single strip?

EDIT: I undid the mod. The same 6-frame strip now properly thumbnails, previews and scans in both Nikon Scan and Vuescan without any issues. 
Side note: I noticed that the scanner showed the "error" blink state on the LED when it got to the end of the strip when the mod was applied (so SA-30), but when I reverted it it didn't blink any errors. So it might be firmware related after all? 

Dump from a successful thumbnail acquisition:

```
***************************************
* Nikon Memory Dumper by Kosma Moczek *
***************************************
Connecting to scanner \\.\usbscan0...
...connected.
Checking model: 0x04b0, product: 0x4001, version: 2.02
...model: DF17811
Checking scanner...
cmd: 12 00 00 00 24 00
...scanner says it's a Nikon   LS-50 ED        2.02
Reading memory at 0xffffc0 to 0xffffdf...
cmd: 3c 02 00 ff ff c0 00 00 20 00
I/O Registers:
0000: ff ff 00 00 ff ff 01 00
0008: ff ff 00 ff ff ff ff eb
0010: ff ff fc 3f ff ff 02 df
0018: 00 ff 00 00 00 00 1f ff

Finished.
Press any key to continue...
```
2-11
Some fiddling:
I found the instruction that retrieves the film door sensor bit, `7ED77360`. Might be enough to patch the bittest out. It tests using BNE/BNQ, but I'm not sure if the actual bit should be set or not. Will do some testing.

It appears 6 times:
- 0x13CD2
- 0x209D6
- 0x21632
- 0x23038
- 0x2FC88
- 0x2FCCA

4-11:
Alas, it's not that easy. I patched out the checks by using ORC and XORC to set the Z flag to 1 or 0 depending on BEQ/BNE but it still doesn't want to eat 6+ frames. If I convert my adapter back to a SA-21, the scanner crashes after the 6th frame. Seems like this is the right track somehow, but I'm at a loss.
The crash was probably more related to Nikon Scan being shitty than the firmware, but still.

**Comment by kosma** [comment]
Fixed in #4.

**Comment by Borislavo** [comment]
I have the same error with LS-40 after applying Kosma's custom firmware and modifyed SA21 at my Win10 x64 PC
